### PR TITLE
Incorporate the 'captured' and 'non-captured' versions of `qml.cond` into `CondCallable`

### PR DESCRIPTION
**Pull request by PietropaoloFrisoni**
**Context:** For 'historical reasons', the captured version  (see sc-66774) and the non-captured version (see sc-69157) of `qml.cond` have a different style, which is rather uneven.

The purpose of this story is to incorporate them in such a way that the implementation is more similar to the one in Catalyst. Or, to make another example, the implementation of `qml.for_loop` in PL (see sc-66736 and sc-69432). The purpose is not to change/add functionalities, but rather to unify the code structure so that it is more elegant.

**Description of the Change:** As above.

**Benefits:** Better and cleaner structure, more similar to Catalyst.

**Possible Drawbacks:** None that I can think of right now directly due to this PR (we just moved some code).

**Related GitHub Issues:** None

**Related Shortcut Stories:** [sc-70342]

### CODE REVIEW

File: pennylane/ops/op_math/condition.py [hunk 1]
```diff
@@ -501,9 +564,6 @@ def qnode(a, x, y, z):
 
         return cond_func
 
-    if qml.capture.enabled():
-        return _capture_cond(condition, true_fn, false_fn, elifs)
-
     if not isinstance(condition, MeasurementValue):
```
Comment (mudit2812) [inline pennylane/ops/op_math/condition.py:567]: The only concern I would've had would've been this line, which specifically avoids using `CondCallable` for MCM conditions. However, with capture enabled, these would be replaced with boolean tracers, so I believe we would still end up using `CondCallable`, and consequently, the conditional primitive, as expected.

I've integrated these changes with #6058 locally and can confirm that it works 😄

Comment (josh146) [reply pennylane/ops/op_math/condition.py:567]: nice! I was also worried about this line but this makes a lot of sense

**Comment by PietropaoloFrisoni** [comment]
@codecov better late than never

**Comment by mudit2812** [comment]
LOL

**Pull request merged**